From: Jeremy Bicha <jeremy.bicha@canonical.com>
Date: Fri, 17 Feb 2023 20:17:51 -0500
Subject: Revert "build: Depend on glib 2.75"

This reverts commit f34c6da9569d71bef7a329294e4d55a65d142875

Revert "Use g_clear_fd() instead of open coding the same behavior"

This reverts commit 49b0a8921cdbb846039847864860884f9fb9ebe4
---
 cogl/cogl/cogl-dma-buf-handle.c            | 4 ++--
 meson.build                                | 2 +-
 src/backends/native/meta-drm-buffer-dumb.c | 4 ++--
 src/wayland/meta-wayland-dma-buf.c         | 6 ++++--
 src/wayland/meta-xwayland.c                | 7 +++++--
 5 files changed, 14 insertions(+), 9 deletions(-)

diff --git a/cogl/cogl/cogl-dma-buf-handle.c b/cogl/cogl/cogl-dma-buf-handle.c
index b8d0929..5b18f10 100644
--- a/cogl/cogl/cogl-dma-buf-handle.c
+++ b/cogl/cogl/cogl-dma-buf-handle.c
@@ -36,7 +36,6 @@
 
 #include <errno.h>
 #include <gio/gio.h>
-#include <glib/gstdio.h>
 #include <linux/dma-buf.h>
 #include <sys/ioctl.h>
 #include <sys/mman.h>
@@ -96,7 +95,8 @@ cogl_dma_buf_handle_free (CoglDmaBufHandle *dmabuf_handle)
   if (dmabuf_handle->destroy_func)
     g_clear_pointer (&dmabuf_handle->user_data, dmabuf_handle->destroy_func);
 
-  g_clear_fd (&dmabuf_handle->dmabuf_fd, NULL);
+  if (dmabuf_handle->dmabuf_fd != -1)
+    close (dmabuf_handle->dmabuf_fd);
 
   g_free (dmabuf_handle);
 }
diff --git a/meson.build b/meson.build
index 26e0aa1..e454c56 100644
--- a/meson.build
+++ b/meson.build
@@ -16,7 +16,7 @@ mutter_builddir = meson.current_build_dir()
 lcms2_req = '>= 2.6'
 colord_req = '>= 1.4.5'
 fribidi_req = '>= 1.0.0'
-glib_req = '>= 2.75.1'
+glib_req = '>= 2.69.0'
 gi_req = '>= 0.9.5'
 graphene_req = '>= 1.10.2'
 gtk3_req = '>= 3.19.8'
diff --git a/src/backends/native/meta-drm-buffer-dumb.c b/src/backends/native/meta-drm-buffer-dumb.c
index 29fb8f1..25d0ed3 100644
--- a/src/backends/native/meta-drm-buffer-dumb.c
+++ b/src/backends/native/meta-drm-buffer-dumb.c
@@ -27,7 +27,6 @@
 
 #include <drm_fourcc.h>
 #include <gio/gio.h>
-#include <glib/gstdio.h>
 #include <xf86drm.h>
 #include <fcntl.h>
 #include <sys/mman.h>
@@ -344,7 +343,8 @@ destroy_dumb_buffer (MetaDrmBufferDumb *buffer_dumb)
   };
   drmIoctl (fd, DRM_IOCTL_MODE_DESTROY_DUMB, &destroy_arg);
 
-  g_clear_fd (&buffer_dumb->dmabuf_fd, NULL);
+  if (buffer_dumb->dmabuf_fd != -1)
+    close (buffer_dumb->dmabuf_fd);
 }
 
 static void
diff --git a/src/wayland/meta-wayland-dma-buf.c b/src/wayland/meta-wayland-dma-buf.c
index 4b44ada..2864866 100644
--- a/src/wayland/meta-wayland-dma-buf.c
+++ b/src/wayland/meta-wayland-dma-buf.c
@@ -40,7 +40,6 @@
 #include "wayland/meta-wayland-dma-buf.h"
 
 #include <drm_fourcc.h>
-#include <glib/gstdio.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
@@ -1790,7 +1789,10 @@ meta_wayland_dma_buf_buffer_finalize (GObject *object)
   int i;
 
   for (i = 0; i < META_WAYLAND_DMA_BUF_MAX_FDS; i++)
-    g_clear_fd (&dma_buf->fds[i], NULL);
+    {
+      if (dma_buf->fds[i] != -1)
+        close (dma_buf->fds[i]);
+    }
 
   G_OBJECT_CLASS (meta_wayland_dma_buf_buffer_parent_class)->finalize (object);
 }
diff --git a/src/wayland/meta-xwayland.c b/src/wayland/meta-xwayland.c
index 8a50b93..28181e2 100644
--- a/src/wayland/meta-xwayland.c
+++ b/src/wayland/meta-xwayland.c
@@ -29,7 +29,6 @@
 #include <errno.h>
 #include <glib-unix.h>
 #include <glib.h>
-#include <glib/gstdio.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
 #include <sys/types.h>
@@ -250,7 +249,11 @@ try_display (int      display,
       g_free (filename);
       filename = NULL;
 
-      g_clear_fd (&fd, NULL);
+      if (fd >= 0)
+        {
+          close (fd);
+          fd = -1;
+        }
     }
 
   *filename_out = filename;
